### PR TITLE
ci: Update release notes tracking issue

### PR DESCRIPTION
**Pull request by tylerbutler**
Adds a GitHub actions workflow to update an issue with the current release notes every time changesets are added or changed on the main branch. This will provide a continually updated snapshot of what the release notes will look like when we release.

I also updated some changeset content to better work with the release note layout. There will be additional changes made in #22067.

### CODE REVIEW

File: fluidBuild.config.cjs [hunk 1]
```diff
@@ -554,11 +554,11 @@ module.exports = {
...
-			"feature": { heading: "✨ New Features" },
-			"tree": { heading: "🌳 SharedTree DDS changes" },
-			"fix": { heading: "🐛 Bug Fixes" },
-			"deprecation": { heading: "⚠️ Deprecations" },
-			"other": { heading: "Other Changes" },
+			feature: { heading: "✨ New Features" },
+			tree: { heading: "🌳 SharedTree DDS changes" },
+			fix: { heading: "🐛 Bug Fixes" },
+			deprecation: { heading: "⚠️ Deprecations" },
+			other: { heading: "Other Changes" },
```
Comment (tylerbutler) [inline fluidBuild.config.cjs:561]:
```suggestion
			"feature": { heading: "✨ New Features" },
			"tree": { heading: "🌳 SharedTree DDS changes" },
			"fix": { heading: "🐛 Bug Fixes" },
			"deprecation": { heading: "⚠️ Deprecations" },
			"other": { heading: "Other Changes" },
```

File: .github/workflows/release-notes-issue.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+name: "Update release notes issue"
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - ".changeset/**"
```
Comment (tylerbutler) [inline .github/workflows/release-notes-issue.yml:7]:
```suggestion
# release-notes-issue.yml
# Updates a release notes tracking issue with the generated release notes.
# The issue to update is set in the RELEASE_NOTES_ISSUE secret.

name: "Update release notes issue"
on:
  push:
    branches:
      - main
    paths:
      - ".changeset/**"
```

File: .changeset/moody-toys-dance.md [hunk 3]
```diff
@@ -2,8 +2,12 @@
 "@fluidframework/tree": minor
 "fluid-framework": minor
 ---
+---
+section: tree
+highlight: true
+---
 
-Allow constructing ArrayNodes from Maps and MapNodes from arrays when unambiguous.
+✨ New! ArrayNodes can now be constructed from Maps, and MapNodes from arrays when unambiguous
```
Comment (alexvy86) [inline .changeset/moody-toys-dance.md:10]: The current wording can be easily red as "when unambiguous" only applying to the second case.
```suggestion
✨ New! When unambiguous, ArrayNodes can now be constructed from Maps and MapNodes from arrays
```

File: .changeset/green-spies-arrive.md [hunk 4]
```diff
@@ -2,22 +2,27 @@
...
+---
+section: tree
+highlight: true
+---
 
-Allow `Record` typed object to be used to construct MapNodes.
+✨ New! `Record`-typed objects can now be used to construct MapNodes
 
-It is now allowed to construct MapNodes from `Record` typed objects, similar to how maps are expressed in JSON.
+IYou can now construct MapNodes from `Record` typed objects, similar to how maps are expressed in JSON.
```
Comment (alexvy86) [inline .changeset/green-spies-arrive.md:12]:
```suggestion
You can now construct MapNodes from `Record` typed objects, similar to how maps are expressed in JSON.
```

File: .github/workflows/release-notes-issue.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
+# release-notes-issue.yml
+# Updates a release notes tracking issue with the generated release notes.
```
Comment (Josmithr) [inline .github/workflows/release-notes-issue.yml:2]: Nit (wording is a bit hard to parse)
```suggestion
# Updates an issue that tracks release notes with the generated release notes.
```

File: .github/workflows/release-notes-issue.yml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      issueNumber:
+        description: 'Issue number to update'
+        default: 22101
+        required: true
+
+permissions:
+  issues: write
+
+jobs:
+  create-issue:
```
Comment (Josmithr) [inline .github/workflows/release-notes-issue.yml:25]: For anyone else who made need to make changes to this in the future, I think more in-depth step-by-step documentation explaining what's going on would be helpful. Especially since most devs on our team don't work with these pipeline templates often.

File: .github/workflows/release-notes-issue.yml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+      - main
+    paths:
+      - ".changeset/**"
+
+  # Allow manually triggering this workflow from the web UI
+  workflow_dispatch:
+    inputs:
+      issueNumber:
+        description: 'Issue number to update'
+        default: 22101
```
Comment (alexvy86) [inline .github/workflows/release-notes-issue.yml:18]: Is this the RELEASE_NOTES_ISSUE? Should this be using the variable as well?

File: .github/workflows/release-notes-issue.yml (hunk 6 above)
Comment (tylerbutler) [reply .github/workflows/release-notes-issue.yml:25]: I added a bunch of comments.

File: .github/workflows/release-notes-issue.yml (hunk 7 above)
Comment (tylerbutler) [reply .github/workflows/release-notes-issue.yml]: Can't use a variable for a default value. I just removed the config option. If we want to change the issue num, we can change the variable itself.

File: .github/workflows/release-notes-issue.yml (hunk 6 above)
Comment (tylerbutler) [reply .github/workflows/release-notes-issue.yml]: I added a bunch of comments.

File: .changeset/many-moments-juggle.md [hunk 8]
```diff
@@ -1,7 +1,12 @@
 ---
 "@fluidframework/container-loader": minor
 ---
+---
+section: deprecation
```
Comment (jzaffiro) [inline .changeset/many-moments-juggle.md:5]: Do we have a defined set of sections that we expect people to use? Maybe we should write them out (or at least, some common examples) in the changesets wiki.

File: .github/workflows/release-notes-issue.yml [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+          body: ${{ steps.relNotes.outputs.content }}
+          edit-mode: replace
+
+      # Prepend the issue intro to the issue.
+      - name: Prepend issue intro
+        uses: julien-deramond/update-issue-body@a7fae45395cac5a23318d38f7b09a58650dfe84f # ratchet:julien-deramond/update-issue-body@v1
+        with:
+          issue-number: ${{ env.ISSUE }}
+          body: ${{ steps.intro.outputs.content }}
+          edit-mode: prepend
```
Comment (alexvy86) [inline .github/workflows/release-notes-issue.yml:101]: Not a blocker, just wondering if there's also an append edit-mode. Feels unusual to replace with the main body then prepend the intro, vs replace with the intro then append the body.

**Review by alexvy86 — APPROVED** [review]

File: .changeset/many-moments-juggle.md (hunk 8 above)
Comment (tylerbutler) [reply .changeset/many-moments-juggle.md:5]: It's fully configurable. When creating changesets the new tool will prompt for the section. Options are dynamic based on the config.

File: .github/workflows/release-notes-issue.yml (hunk 9 above)
Comment (tylerbutler) [reply .github/workflows/release-notes-issue.yml:101]: I could do it that way - but it makes the diffs in the issue history a little less useful. Not a big deal. I'll rearrange the logic and append the main body.

File: .changeset/many-moments-juggle.md (hunk 8 above)
Comment (alexvy86) [reply .changeset/many-moments-juggle.md:5]: Yes, Tyler added them [as config in the root fluidBuild.config.js](https://github.com/microsoft/FluidFramework/blob/5504a9f6470d732861770c2ebb610687fc01875e/fluidBuild.config.cjs#L555-L563) recently. I agree adding a link to those in the wiki would be nice. But also note that if people generate changesets with `pnpm changeset` (vs manually copy pasting an existing one), the tool will give you the options based on that config. If I've been understanding the recent changes correctly 😄 .

File: .github/workflows/release-notes-issue.yml (hunk 9 above)
Comment (alexvy86) [reply .github/workflows/release-notes-issue.yml:101]: Oh, I see why that would be... I don't mind the prepend that much to be honest, so a comment explaining that's the reason would be good enough, IMO. I leave it up to you.
